### PR TITLE
prototypefactory refactor

### PR DESCRIPTION
**Pull request by wpride**

### CODE REVIEW

**Comment by wpride** [comment]
cross request https://github.com/dimagi/javarosa/pull/126/files

File: app/src/org/commcare/util/externalizable/ImprovedPrototypeFactory.java [hunk 1]
```diff
@@ -60,4 +35,8 @@ public Class getClass (byte[] hash) {
         return prototypes.get(getHash(hash));
     }
 
+    public void storeHash(Class c, byte[] hash){
```
Comment (ctsims) [inline app/src/org/commcare/util/externalizable/ImprovedPrototypeFactory.java]: I don't think this is ever getting called

File: app/src/org/commcare/util/externalizable/ImprovedPrototypeFactory.java [hunk 2]
```diff
@@ -1,51 +1,26 @@
...
-        }
-        
-        prototypes.put(getHash(hash), c);
+    protected void lazyInit() {
+        initialized = false;
+        prototypes = new Hashtable<Integer, Class>();
+        super.lazyInit();
     }
     
     private Integer getHash(byte[] hash) {
```
Comment (ctsims) [inline app/src/org/commcare/util/externalizable/ImprovedPrototypeFactory.java]: Also not getting called

File: app/src/org/commcare/util/externalizable/ImprovedPrototypeFactory.java [hunk 3]
```diff
@@ -1,51 +1,26 @@
 package org.commcare.util.externalizable;
 
-import java.util.Hashtable;
-
 import org.javarosa.core.util.PrefixTree;
```
Comment (ctsims) [inline app/src/org/commcare/util/externalizable/ImprovedPrototypeFactory.java]: Can you add @Override to everything in this class that is overriding child behavior

**Comment by phillipm** [comment]
@wpride can we close this PR? Seems like all of its commits are also present in https://github.com/dimagi/commcare-odk/pull/488

**Pull request closed without merging**